### PR TITLE
[G-72] Authenticate with gateway instead of central registry

### PR DESCRIPTION
**Pull request by ganesanarun**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  1
- src/main/java/in/projecteka/consentmanager/clients/ServiceAuthenticationClient.java  1
- src/main/java/in/projecteka/consentmanager/common/ServiceAuthentication.java  1
- src/main/java/in/projecteka/consentmanager/common/GatewayTokenVerifier.java  3
         

Complexity decreasing per file
==============================
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -1
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  4
         

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  -6
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5752579&bid=18831425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  1
- src/main/java/in/projecteka/consentmanager/common/ServiceAuthentication.java  1
- src/main/java/in/projecteka/consentmanager/clients/ServiceAuthenticationClient.java  1
- src/main/java/in/projecteka/consentmanager/common/GatewayTokenVerifier.java  3
         

Complexity decreasing per file
==============================
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -1
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  4
         

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  -6
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5752579&bid=18831425)

File: src/main/java/in/projecteka/consentmanager/SecurityConfiguration.java [hunk 1]
```diff
@@ -144,9 +144,9 @@ public ReactiveAuthenticationManager authenticationManager() {
 
     @Bean
     public Authenticator authenticator(@Qualifier("identityServiceJWKSet") JWKSet jwkSet,
-                                       CacheAdapter<String, String> blacklistedTokens,
+                                       CacheAdapter<String, String> blockListedTokens,
```
Comment (dheeraj-p) [inline src/main/java/in/projecteka/consentmanager/SecurityConfiguration.java:147]: `blockListedTokens` ---> `blackListedTokens`

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/ServiceAuthentication.java  1
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  1
- src/main/java/in/projecteka/consentmanager/clients/ServiceAuthenticationClient.java  1
- src/main/java/in/projecteka/consentmanager/common/GatewayTokenVerifier.java  3
         

Complexity decreasing per file
==============================
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -1
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  4
         

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -5
+ src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5752579&bid=18831425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/GatewayTokenVerifier.java  3
- src/main/java/in/projecteka/consentmanager/clients/ServiceAuthenticationClient.java  1
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  1
- src/main/java/in/projecteka/consentmanager/common/ServiceAuthentication.java  1
         

Complexity decreasing per file
==============================
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -1
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  4
         

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -5
+ src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5752579&bid=18831425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/ServiceAuthentication.java  1
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  1
- src/main/java/in/projecteka/consentmanager/common/GatewayTokenVerifier.java  3
- src/main/java/in/projecteka/consentmanager/clients/ServiceAuthenticationClient.java  1
         

Complexity decreasing per file
==============================
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -1
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  4
         

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -5
+ src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5752579&bid=18831425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/clients/ServiceAuthenticationClient.java  1
- src/main/java/in/projecteka/consentmanager/common/GatewayTokenVerifier.java  3
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  1
- src/main/java/in/projecteka/consentmanager/common/ServiceAuthentication.java  1
         

Complexity decreasing per file
==============================
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -1
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/common/GatewayTokenVerifierTest.java  4
         

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/link/link/LinkUserJourneyTest.java  -5
+ src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5752579&bid=18831425)

**Pull request merged**